astropy/coordinates/baseframe.py

```diff
@@ -412,9 +485,39 @@ def representation(self):
             self._representation = self.default_representation
         return self._representation
 
+    @representation_cls.setter
+    def representation_cls(self, value):
+        self._representation = _get_repr_cls(value)
+
+    @property
+    def representation(self):
+        """
+        The representation of the data in this frame, as a class that is
+        subclassed from `~astropy.coordinates.BaseRepresentation`.  Can
+        also be *set* using the string name of the representation.
+        """
+        # TODO: in the future, we may want to raise a deprecation warning here
+        return self.representation_cls
+
     @representation.setter
     def representation(self, value):
-        self._representation = _get_repr_cls(value)
+        # TODO: in the future, we may want to raise a deprecation warning here
+        self.representation_cls = value
+
+    @property
+    def differential_cls(self):
+        """
+        The differential representation of the velocity data in this frame, as a
+        class that is subclassed from `~astropy.coordinates.BaseDifferential`.
+        Can also be *set* using the string name of the differential.
+        """
+        if not hasattr(self, '_differential'):
+            self._differential = self.default_differential
+        return self._differential
+
+    @differential_cls.setter
+    def differential_cls(self, value):
+        self._differential = _get_diff_cls(value)
```

Comment:
Doesn't the differential class have to be consistent with that of the representation?